File: Programmers/Sliding Window/가장 긴 팰린드롬.cpp
#include <algorithm>
#include <string>

using namespace std;

int getPalinedrome(string s, int left, int right) {
    while(left >= 0 && right < s.size() && s[left] == s[right]) {
        left--;
        right++;
    }
    return right - left - 1;
}

int solution(string s)
{
    int answer=0;
    for (int i = 0; i < s.size(); i++) {
        int len1 = getPalinedrome(s, i, i); // i가 중심인 경우
        int len2 = getPalinedrome(s, i, i + 1); // i와 i + 1이 중심인 경우
        answer = max({answer, len1, len2});
    }

    return answer;
}

File: Programmers/Sliding Window/가장 긴 팰린드롬.py
def getPaline(s, left, right, n):
    while 0 <= left and right < n and s[left] == s[right]:
        left -= 1
        right += 1
    return right - left - 1 # while 내부가 실행되고 조건이 끝나므로 -1해야

def solution(s):
    answer = 0
    n = len(s)
    for i in range(n):
        answer = max(answer, getPaline(s, i, i, n)) # 홀수 길이
        answer = max(answer, getPaline(s, i, i + 1, n)) # 짝수 길이

    return answer